venom/protocol/__init__.py

```diff
@@ -40,12 +45,12 @@ def __new__(cls, fmt: Type[Message], field_names: Set[str] = None):
         #     return Protocol.__new__(cls, Empty)
         else:
             instance = super(Protocol, cls).__new__(cls)
-            instance.__init__(fmt, field_names)
+            instance.__init__(fmt, field_names_)
             return instance
 
     def __init__(self, fmt: Type[Message], field_names_: Set[str] = None):
         if field_names_ is None:
-            field_names_ = field_names(fmt)
+            field_names_ = chain(field_names(fmt), (f.name for f in fields(fmt)))
```

Comment:
Sorry, what is this doing? `field_names(fmt)` should be equal to `[f.name for field in fields fields(fmt)]`. If it is not, you have to fix `field_names()`. A unit test is in order.